src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -200,11 +202,19 @@ static Object preProcessValue(Object value, Schema schema, Schema newSchema) {
         }
         return result;
       case MAP:
-        keySchema = schema.keySchema();
-        valueSchema = schema.valueSchema();
-        ArrayList<Struct> mapStructs = new ArrayList<>();
-        Map<?, ?> map = (Map<?, ?>) value;
+        Schema keySchema = schema.keySchema();
         Schema newValueSchema = newSchema.valueSchema();
+        Schema valueSchema = schema.valueSchema();
+        Map<?, ?> map = (Map<?, ?>) value;
+        if (keySchema.type() == Schema.Type.STRING) {
+          Map<Object, Object> processedMap = new HashMap<>();
+          for (Map.Entry<?, ?> entry: map.entrySet()) {
+            processedMap.put(preProcessValue(entry.getKey(), keySchema, newSchema.keySchema()),
+                    preProcessValue(entry.getValue(), valueSchema, newValueSchema));
+          }
+          return processedMap;
+        }
+        ArrayList<Struct> mapStructs = new ArrayList<>();
```

Comment:
nit: Ideally it'd be `List<Struct>` (for instance in the test we cast to List anyway). If you choose to fix this, pls fix the declaration of `ArrayList<Object>` in the ARRAY case as well.